src/combobox/combobox.component.ts

```diff
@@ -469,9 +475,14 @@ export class ComboBox implements OnChanges, AfterViewInit, AfterContentInit {
 		event.stopPropagation();
 		event.preventDefault();
 
-		this.input.nativeElement.value = "";
-
 		this.clearSelected();
+
+		// This makes sure there are no selected items before clearing the input manually.
+		// The view needs to be updated manually because the input will not clear automatically
+		// if we clear the selected item for a single selection combobox.
+		const selected = this.view.getSelected();
+		this.input.nativeElement.value = selected ? selected : "";
```

Comment:
This seems redundant since two lines above we call `clearSelected` which ... clears the selection

Reply:
It clears the selection and the content does update correctly but the view does not update, it keeps the same value in the input (even if the model does not reflect this).

Reply:
But that's not what this code is doing ... this code is checking if there's a selection and then not clearing the input value

Reply:
There should be no way that there is a selection at this point (barring disabled items which may not even be part of the `getSelected` result set)